### PR TITLE
Simple Consul KV

### PR DESCRIPTION
**Pull request by debovema**

### CODE REVIEW

File: consul/consul.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+func init() {
+	logger := log.RootLogger()
+
+	client, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		logger.Error("Unable to initialize Consul property resolver for configuration management.")
+	}
+
+	kv = client.KV()
```
Comment (ffassler) [inline consul/consul.go]: The variable _client_ should not be used if an error occured when creating client :

`client, err := api.NewClient(api.DefaultConfig())`

File: consul/consul.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	"github.com/hashicorp/consul/api"
+	"github.com/project-flogo/core/data/property"
+	"github.com/project-flogo/core/engine"
+	"github.com/project-flogo/core/support/log"
+	"strings"
+)
+
+var (
+	kv     *api.KV
```
Comment (ffassler) [inline consul/consul.go]: The variable _kv_ should be in the  struct _SimpleConsulKVValueResolver_

File: property-resolver-consul/consul.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// Resolve property value from a Consul KV Store
+type SimpleConsulKVValueResolver struct {
+}
+
+func (resolver *SimpleConsulKVValueResolver) LookupValue(key string) (interface{}, bool) {
+	key = strings.Replace(key, ".", "/", -1)
+
+	consul_key := "flogo/" + engine.GetAppName() + "/" + key
```
Comment (ffassler) [inline property-resolver-consul/consul.go]: Camel case naming ? _consulKey_

**Review by ffassler — CHANGES_REQUESTED** [review]

File: consul/consul.go (hunk 1 above)
Comment (debovema) [reply consul/consul.go]: Resolved by https://github.com/square-it/flogo-config-mgmt/pull/2/commits/b60b05acd9a1ff692520b8232dd6cdf88f880965

File: consul/consul.go (hunk 2 above)
Comment (debovema) [reply consul/consul.go]: Resolved by https://github.com/square-it/flogo-config-mgmt/pull/2/commits/b60b05acd9a1ff692520b8232dd6cdf88f880965

File: property-resolver-consul/consul.go (hunk 3 above)
Comment (debovema) [reply property-resolver-consul/consul.go]: Resolved by https://github.com/square-it/flogo-config-mgmt/pull/2/commits/b60b05acd9a1ff692520b8232dd6cdf88f880965

**Review by ffassler — APPROVED** [review]

**Pull request merged**